File: Min-Stack/MinStack.java
import java.util.*;

class MinStack {
    private Stack<Integer> stack;
    private Stack<Integer> minStack;

    MinStack() {
        stack = new Stack<Integer>();
        minStack = new Stack<Integer>();
    }

    public void push(int x) {
        stack.push(x);
        if (minStack.empty() || minStack.peek() >= x) {
            minStack.push(x);
        }
        return;
    }

    public void pop() {
        if ((int)stack.peek() == (int)minStack.peek()) {
            minStack.pop();
        }
        stack.pop();
        return;
    }

    public int top() {
        return stack.peek();
    }

    public int getMin() {
        return minStack.peek();
    }
}

File: Min-Stack/MinStack.cpp
#include <stack>

using namespace std;

class MinStack {
    stack<int> elements;
    stack<int> minStack;

public:
    void push(int x) {
        elements.push(x);
        if (minStack.empty() || minStack.top() >= x) {
            minStack.push(x);
        }
        return;
    }

    void pop() {
        if (elements.top() <= minStack.top()) {
            minStack.pop();
        }
        elements.pop();
        return;
    }

    int top() {
        return elements.top();
    }

    int getMin() {
        return minStack.top();
    }
};

int main() {
  return 0;
}
